### PR TITLE
KMP in C

### PR DESCRIPTION
**Pull request by shreyakapoor08**
Fixes #2555 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
KMP in C added

#### Languages Used:
C 

#### Files Added:
KMP.c 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by shreyakapoor08** [comment]
@MastersAbh Hey, I just checked it once again and it has 4 space indentation only.. Kindly check

**Comment by shreyakapoor08** [comment]
@MastersAbh Kindly check now

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+///Sample IP:
+// BACPARKWRPARM
+// PAR
+
+///Sample OP:
+// Pattern occurs with shift 3
+// Pattern occurs with shift 9
+
```
Comment (somya-kapoor) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:71]: Remove the line 71,72,73

**Comment by shreyakapoor08** [comment]
@somya-kapoor Done. Kindly Review

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return 0;
+}
+
+///Sample IP:
+// BACPARKWRPARM
+// PAR
+
+///Sample OP:
+// Pattern occurs with shift 3
+// Pattern occurs with shift 9
```
Comment (MastersAbh) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:70]: Add one blank line at the end of the file.

**Comment by shreyakapoor08** [comment]
@MastersAbh Kindly check

**Pull request merged**